lib/manageiq/providers/workflows/builtin_methods.rb

```diff
@@ -21,6 +21,41 @@ def self.email(params = {}, _secrets = {}, _context = {})
 
         private_class_method def self.email_cleanup(*)
         end
+
+        def self.provision_execute(_params = {}, _secrets = {}, context = {})
+          return unless context.execution["_object_type"] == "ServiceTemplateProvisionTask"
+
+          ::MiqRequestTask.find(context.execution["_object_id"])&.execute
+
+          {"miq_request_task_id" => request_task_id}
+        end
+
+        private_class_method def self.provision_execute_status!(runner_context)
+          request_task = ::MiqRequestTask.find_by(:id => runner_context["miq_request_task_id"])
+          return if request_task.nil?
+
+          case request_task.statemachine_task_status
+          when "error"
+            reason = task.message.sub(/^Error: /, "")
+            runner_context["running"] = false
+            runner_context["success"] = false
+            runner_context["output"] = {"Error" => "States.TaskFailed", "Cause" => reason}
+          when "retry"
+            runner_context["running"] = false
```

Comment:
I'm not sure what retry implies, but I think it should still be "running"?

Reply:
Oh yeah good point, does "retry" here mean we want the ASL workflow to retry or it is the backend doing the retry?

Reply:
I was thinking that it was the same thing as a failure, the difference being that we will revisit this.

see how this is written in the floe:
https://github.com/ManageIQ/manageiq-content/pull/746

I agree, this suggests that it is a retry, but this version of `retry` may be more like a `wait`.

Reply:
confused by this.
agree with `running=true`, but it is not complete. it is basically a `wait` state so then `success` is not fully determined.

Reply:
> agree with running=true, but it is not complete.

Do you mean you don't think we should set success=true?

This could apply to all of these, but in general the container_runners don't set success=true/false until after it has completed so if that's what you mean then yes I agree but then still not sure why we'd say running=false,success=false because that is an error case.

Reply:
I was thinking running=true and success is not yet determined (not sure if that should be false or just not set at all?)